analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompiler.cs

```diff
@@ -92,6 +92,7 @@ public SyntaxNode GetMethodDeclaration(string typeDotMethodName)
         public INamespaceSymbol GetNamespaceSymbol(string name)
         {
             var symbol = GetNodes<CSharpSyntax.NamespaceDeclarationSyntax>()
+                .Concat<SyntaxNode>(GetNodes<CSharpSyntax.FileScopedNamespaceDeclarationSyntax>())
```

Comment:
```suggestion
            var symbol = GetNodes<CSharpSyntax.BaseNamespaceDeclarationSyntax>()
```

You can use the base class to retrieve both syntax kinds.